### PR TITLE
[net] Added SetSocketNoDelay() utility function

### PR DESCRIPTION
**Pull request by tjps**
De-duplicated a small bit of systems code

### CODE REVIEW

File: src/netbase.cpp [hunk 1]
```diff
@@ -728,6 +724,19 @@ bool SetSocketNonBlocking(SOCKET& hSocket, bool fNonBlocking)
     return true;
 }
 
+bool SetSocketNoDelay(SOCKET& hSocket)
+{
+    int set = 1;
+    int rc = setsockopt(hSocket, IPPROTO_TCP, TCP_NODELAY,
+#ifdef WIN32
```
Comment (paveljanik) [inline src/netbase.cpp]: I don't like this ;-)

Comment (tjps) [inline src/netbase.cpp]: You mean in the middle of the function call?  I agree it looks visually worse, but I like deduplicating code as much as possible.

However on longer consideration, I realized that the conversion from (const char*) to (const void*) will happen implicitly on any platform, so this ifdef isn't even needed.

**Comment by sipa** [comment]
utACK

File: src/netbase.cpp [hunk 2]
```diff
@@ -728,6 +724,13 @@ bool SetSocketNonBlocking(SOCKET& hSocket, bool fNonBlocking)
     return true;
 }
 
+bool SetSocketNoDelay(SOCKET& hSocket)
+{
+    int set = 1;
+    int rc = setsockopt(hSocket, IPPROTO_TCP, TCP_NODELAY, (const char*)&set, sizeof(int));
```
Comment (laanwj) [inline src/netbase.cpp]: Indeed, seems like the WIN32 check was unnecessary, every pointer can be passed to a `void*` parameter including `char*` so the redundant cast doesn't matter.

File: src/netbase.cpp [hunk 3]
```diff
@@ -728,6 +724,13 @@ bool SetSocketNonBlocking(SOCKET& hSocket, bool fNonBlocking)
...
+    return rc != -1;
```
Comment (laanwj) [inline src/netbase.cpp]: I'd feel slightly better with `rc == 0`. 0 is guaranteed to be the success value both in Windows and UNIX, whereas on failure [windows returns](https://msdn.microsoft.com/en-us/library/windows/desktop/ms740476(v=vs.85).aspx) `SOCKET_ERROR` (which may or may not be -1, it's not mentioned).

Comment (tjps) [inline src/netbase.cpp]: But of course, I usually idiomatically check rc == 0 from system calls.  Not sure why I went with != -1.

**Comment by tjps** [comment]
bump on an easy dedupe merge

**Comment by TheBlueMatt** [comment]
@theuni ping Seems this stuff could generally use a cleanup (merge SetSocketNonBlocking/SetSocketNoDelay, also do we need to set SO_NOSIGPIPE on the newly-listened socket (looks like it inherits), maybe would be nice to merge that too?

Anyway, utACK ad415bc16afdb4fdb6d619624762f1324751a824 this itself, its still a strict improvement.

**Comment by theuni** [comment]
utACK ad415bc16afdb4fdb6d619624762f1324751a824.

@TheBlueMatt We're about to get new socket creation handling from libevent (I'm working on it now), so I'm not too worried about further refactors here.

**Pull request merged**